src/Neo.VM/JumpTable/JumpTable.Splice.cs

```diff
@@ -95,13 +95,22 @@ public virtual void SubStr(ExecutionEngine engine, Instruction instruction)
         {
             int count = (int)engine.Pop().GetInteger();
             if (count < 0)
-                throw new InvalidOperationException($"The value {count} is out of range.");
+                throw new InvalidOperationException($"The value of count {count} is out of range.");
+
             int index = (int)engine.Pop().GetInteger();
             if (index < 0)
-                throw new InvalidOperationException($"The value {index} is out of range.");
+                throw new InvalidOperationException($"The value of index {index} is out of range.");
+
             var x = engine.Pop().GetSpan();
-            if (index + count > x.Length)
-                throw new InvalidOperationException($"The value {count} is out of range.");
+            if (count > x.Length)
+                throw new InvalidOperationException($"The value of count {count} is out of range.");
+
+            if (index >= x.Length)
+                throw new InvalidOperationException($"The value of index {index} is out of range.");
+
+            if (checked(index + count) > x.Length)
```

Comment:
```suggestion
            if (unchecked(index + count) > x.Length)
```

This would be better. If it overflows it will be bigger than the length. This way we don't have dotnet generic `overflow` exception.

Reply:
> This would be better. If it overflows it will be bigger than the length. This way we don't have dotnet generic `overflow` exception.

`checked` is same as `Memcpy`

Reply:
Maybe we need a HF